### PR TITLE
<DeveloperModeProvider> and useTokenList completed

### PR DESCRIPTION
**Pull request by marekkirejczyk**

### CODE REVIEW

File: src/app/context/token_list.tsx
Comment (akonior) [inline src/app/context/token_list.tsx:1]: should be '.ts' file type instead of '.tsx'

File: src/app/context/developer_mode.tsx [hunk 1]
```diff
@@ -1,35 +1,39 @@
 import {
   AccountManager,
   AztecAddress,
```
Comment (akonior) [inline src/app/context/developer_mode.tsx:3]: Unused import

File: src/app/context/token_list.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    readonly address: AztecAddress;
+    readonly decimals: number;
+    readonly logoURI?: string;
+  }
+
+export const TOKEN_LIST = [
+  {
+    name: 'Ether',
+    symbol: 'ETH',
+    address: AztecAddress.fromString(TOKEN_CONTRACT_ADDRESS_AS_STRING),
```
Comment (akonior) [inline src/app/context/token_list.tsx:18]: Use `TOKEN_CONTRACT_ADDRESS`

File: src/app/home.tsx [hunk 3]
```diff
@@ -4,18 +4,18 @@ import { UserAccount } from './model/UserAccount.js';
 import { CreateAccount } from './modals/CreateAccount.js';
 import { Main } from './modals/Main.js';
 import { TokenContract } from './account/token.js';
+import { useTokenList } from './context/token_list.js';
```
Comment (akonior) [inline src/app/home.tsx:7]: Unused imports

File: src/app/context/developer_mode.tsx [hunk 4]
```diff
@@ -1,35 +1,39 @@
...
     getSchnorrAccount(pxe, encryptionKey, INITIAL_SANDBOX_SIGNING_KEYS[i], INITIAL_SANDBOX_SALTS[i]),
   );
 }
 
-const TOKEN_CONTRACT_ADDRESS = '0x2f45f498b7912c779dde8e3594622e36d7908088b09e99ab91caaafb40d1f9ef';
 
 export function DeveloperModeProvider({ children }: { children: ReactNode }) {
   const { pxe } = usePXE();
   const adminAccount = getSandboxAccounts(pxe)[0];
+  const tokenContractAddress = useTokenList()[0].address;
```
Comment (akonior) [inline src/app/context/developer_mode.tsx:32]: This is minor but I would just use `TOKEN_LIST[0].address`, I don't see the advantage of using hooks for constants. Hooks implies that there is mutable state underneath. But it doesn't bother me much.

Comment (marekkirejczyk) [reply src/app/context/developer_mode.tsx:32]: I assume in the future there will be (mutable?) state, i.e. list will be stored on chain, or have properties that are fetched from chain.

Comment (marekkirejczyk) [reply src/app/context/developer_mode.tsx:32]: ✅ Ok, you convinced me, done.

File: src/app/context/token_list.tsx (hunk 2 above)
Comment (marekkirejczyk) [reply src/app/context/token_list.tsx:18]: ✅

File: src/app/context/developer_mode.tsx (hunk 1 above)
Comment (marekkirejczyk) [reply src/app/context/developer_mode.tsx:3]: ✅

File: src/app/context/token_list.tsx
Comment (marekkirejczyk) [reply src/app/context/token_list.tsx:1]: ✅

**Review by akonior — APPROVED** [review]

**Pull request merged**